tests/kw_test.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+. ./kw.sh --source-only
+
+function suite
+{
+  suite_addTest "testHelp"
+  suite_addTest "testVariables"
+  suite_addTest "testExported"
+}
+
+function testHelp
+{
+  HELP_OUTPUT=`kw help | head -n 1`
+  assertTrue "Help text not displaying correctly." '[[ $HELP_OUTPUT =~ Usage:\ kw.* ]]'
```

Comment:
Here we have a problem that is equally valid for all your current implementation: the invoking of "kw" function. This is a problem because you can use the "kw" installed in your environment instead of using the one present in the repository (remember, you want to test the code in the repo).  One solution to avoid the mix between the environments could be:
```bash
function testHelp
{
  (
    unset kw
    . ./kw.sh --source-only
    HELP_OUTPUT=`kw help | head -n 1`
    [[ $HELP_OUTPUT =~ Usage:\ kw.* ]]
    assertTrue "Help text not displaying correctly." $?
  )
}
```

creates a subshell that starts as a copy of the current shell, because of this we unset kw in the beginning and then load it inside the subshell. This approach isolates the environment, but you have to test/check/improve it.
Additionally, for some reason, your assertTrue condition always fail on my computer. Take a look at the fix that I proposed in the above code.